ledger/ledger-utxo/src/main/kotlin/net/corda/v5/ledger/utxo/TransactionComponents.kt

```diff
@@ -0,0 +1,196 @@
+     * @return Returns true if the current [TimeWindow] contains the specified [Instant]; otherwise, false.
+     */
+    operator fun contains(instant: Instant): Boolean
+}
+
+/**
+ * Defines information that applies to a transaction state.
+ *
+ * @property contractId The class name of the [Contract] associated with the transaction state.
+ * @property notary The [Party] of the notary associated with the transaction state.
+ * @property encumbrance The index of an associated, encumbered state, or null if no encumbrance applies to the associated transaction state.
+ * @property constraint The [CpkConstraint] associated with the transaction state.
+ */
+@DoNotImplement
+@CordaSerializable
+interface TransactionStateInformation {
+    val contractId: String
+    val notary: Party
+    val encumbrance: Int?
+    val constraint: CpkConstraint
+}
+
+/**
+ * Defines a transaction state, composed of a [ContractState] and associated [TransactionStateInformation].
+ *
+ * @param T The underlying type of the [ContractState] instance.
+ * @property contractState The [ContractState] component of the current [TransactionState] instance.
+ * @property information The [TransactionStateInformation] component of the current [TransactionState] instance.
+ */
+@DoNotImplement
+@CordaSerializable
+interface TransactionState<out T : ContractState> {
+    val contractState: T
+    val information: TransactionStateInformation
+}
+
+/**
+ * Represents a composition of a [TransactionState] and a [StateRef].
+ *
+ * @property state The [TransactionState] component of the current [StateAndRef].
+ * @property ref The [StateRef] component of the current [StateAndRef].
+ */
+@DoNotImplement
+@CordaSerializable
+interface StateAndRef<out T : ContractState> {
+    val state: TransactionState<T>
+    val ref: StateRef
+}
+
+/**
+ * Defines a reference to a [ContractState].
+ *
+ * @property index The index of the state in the transaction's outputs in which the referenced state was created.
+ * @property transactionHash The hash of the transaction in which the referenced state was created.
+ */
+@DoNotImplement
+@CordaSerializable
+interface StateRef {
+    val index: Int
+    val transactionHash: SecureHash
+}
+
+/**
+ * Represents input and output states that are grouped by a key that is common to all inputs and outputs.
+ *
+ * @constructor Creates a new instance of the [InputOutputGroup] data class.
+ * @param T The underlying type of the grouped [ContractState] instances.
+ * @param K The underlying type of the grouping key.
+ * @property inputs The inputs that are grouped by the specified grouping key.
+ * @property outputs The outputs that are grouped by the specified grouping key.
+ * @property groupingKey The grouping key that is common to all grouped input and output states.
+ */
+@DoNotImplement
+@CordaSerializable
+interface InputOutputGroup<out T : ContractState, out K : Any> {
```

Comment:
Does anyone use this?

Reply:
@mnesbit Not that I know of. Delete?
What I think would be more valuable for `UtxoLedgerTransaction` and `UtxoTransactionBuilder` is a data structure and associated functions to add or get grouped encumbered states, but again, maybe something that isn't required for beta.